### ISSUE TITLE
KeyError: 'plan_detected' in run_sim()

### ISSUE THREAD

**Issue by deanthedream** [dsavransky/EXOSIMS#73]
When running

```
import sys, os.path, EXOSIMS, EXOSIMS.MissionSim
folder = os.path.normpath(os.path.expandvars('$HOME/Documents/exosims/EXOSIMS/EXOSIMS/Scripts'))
filename = 'sS_AYO3.json'
#filename = 'sS_intTime6_KeplerLike2.json'
scriptfile = os.path.join(folder,filename)
sim = EXOSIMS.MissionSim.MissionSim(scriptfile)
sim.run_sim()
```
From the dean-dev branch with the attached scriptfile,
I get the following error:

```
KeyError                                  Traceback (most recent call last)
/home/dean/Documents/exosims/EXOSIMS/EXOSIMS/Scripts/quickrun.py in <module>()
      5 scriptfile = os.path.join(folder,filename)
      6 sim = EXOSIMS.MissionSim.MissionSim(scriptfile)
----> 7 sim.run_sim()

/home/dean/Documents/exosims/EXOSIMS/EXOSIMS/MissionSim.py in run_sim(self)
    229         """
    230 
--> 231         res = self.SurveySimulation.run_sim()
    232 
    233         return res

/home/dean/Documents/exosims/EXOSIMS/EXOSIMS/Prototypes/SurveySimulation.py in run_sim(self)
    297                 if TK.currentTimeNorm > TK.missionLife and len(self.starExtended) == 0:
    298                     for i in range(len(self.DRM)):
--> 299                         if np.any([x == 1 for x in self.DRM[i]['plan_detected']]):
    300                             self.starExtended = np.unique(np.append(self.starExtended,
    301                                     self.DRM[i]['star_ind']))

KeyError: 'plan_detected'
> /home/dean/Documents/exosims/EXOSIMS/EXOSIMS/Prototypes/SurveySimulation.py(299)run_sim()
    297                 if TK.currentTimeNorm > TK.missionLife and len(self.starExtended) == 0:
    298                     for i in range(len(self.DRM)):
--> 299                         if np.any([x == 1 for x in self.DRM[i]['plan_detected']]):
    300                             self.starExtended = np.unique(np.append(self.starExtended,
    301                                     self.DRM[i]['star_ind']))

ipdb> 
```

This appears to indicate that no 'plan_detected ' flags are created by run_sim().
I circumvented this issue by creating my own run_sim() in older versions of starkAYO.

**Comment by dsavransky** [comment]
This is in fact a bug, since the planet_detected keyword was replaced with det_status a long time ago.  It's not clear how you hit it though, since you'd have to be entering this code at a time after the end of the mission time.  Perhaps you were running after the completion of a simulation without reseting timekeeping?

**Comment by ChrisDelaX** [comment]
This is a hidden bug, that was not happening so far because no one was
using the starExtended array.
I believe the bugfix should be straightforward. It only concerns
SurveySimulation and starkAYO:
```
nat-oitwireless-inside-vapornet100-b-4963:gitexosims cdelacroix$ grep -rn
"plan_det" ./ --include="*.py"
.//EXOSIMS/Prototypes/SurveySimulation.py:299:                        if
np.any([x == 1 for x in self.DRM[i]['plan_detected']]):
.//EXOSIMS/SurveySimulation/starkAYO.py:295:                            if
np.any([x == 1 for x in self.DRM[i]['plan_detected']]):
```



2017-09-05 14:54 GMT-04:00 Dmitry Savransky <notifications@github.com>:

**Comment by dsavransky** [comment]
@deanthedream confirm that this is closed in 1.4 prerelase.

**Comment by deanthedream** [comment]
We removed extendedn mission elements from TK and SS
